internal/services/formatters/generic/dependency_check/formatter.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencycheck
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/ZupIT/horusec-devkit/pkg/entities/vulnerability"
+	"github.com/ZupIT/horusec-devkit/pkg/enums/languages"
+	"github.com/ZupIT/horusec-devkit/pkg/enums/tools"
+	"github.com/ZupIT/horusec-devkit/pkg/utils/logger"
+
+	dockerEntities "github.com/ZupIT/horusec/internal/entities/docker"
+	"github.com/ZupIT/horusec/internal/enums/images"
+	"github.com/ZupIT/horusec/internal/helpers/messages"
+	"github.com/ZupIT/horusec/internal/services/formatters"
+	dependencyCheckEntities "github.com/ZupIT/horusec/internal/services/formatters/generic/dependency_check/entities"
+	vulnhash "github.com/ZupIT/horusec/internal/utils/vuln_hash"
+)
+
+type Formatter struct {
+	formatters.IService
+}
+
+func NewFormatter(service formatters.IService) formatters.IFormatter {
+	return &Formatter{
+		service,
+	}
+}
+
+func (f *Formatter) StartAnalysis(projectSubPath string) {
+	if f.ToolIsToIgnore(tools.OwaspDependencyCheck) || f.IsDockerDisabled() || f.IsOwaspDependencyCheckDisable() {
+		logger.LogDebugWithLevel(messages.MsgDebugToolIgnored + tools.OwaspDependencyCheck.ToString())
+		return
+	}
+
+	f.SetAnalysisError(f.startDependencyCheck(projectSubPath), tools.OwaspDependencyCheck, projectSubPath)
+	f.LogDebugWithReplace(messages.MsgDebugToolFinishAnalysis, tools.OwaspDependencyCheck, languages.Generic)
+}
+
+func (f *Formatter) startDependencyCheck(projectSubPath string) error {
+	f.LogDebugWithReplace(messages.MsgDebugToolStartAnalysis, tools.OwaspDependencyCheck, languages.Generic)
+
+	output, err := f.ExecuteContainer(f.getConfigData(projectSubPath))
+	if err != nil {
+		return err
+	}
+
+	return f.parseOutput(output)
+}
+
+func (f *Formatter) getConfigData(projectSubPath string) *dockerEntities.AnalysisData {
+	analysisData := &dockerEntities.AnalysisData{
+		CMD:      f.AddWorkDirInCmd(CMD, projectSubPath, tools.OwaspDependencyCheck),
+		Language: languages.Generic,
+	}
+
+	return analysisData.SetData(f.GetCustomImageByLanguage(languages.Generic), images.Generic)
+}
+
+func (f *Formatter) parseOutput(output string) error {
+	var analysis *dependencyCheckEntities.Analysis
+
+	index := strings.Index(output, "{")
```

Comment:
The output is not a entire json? Why we  need to find the index of `{`?

Reply:
Json comes with a text at the beginning, the index is used to remove this text.